src/seismometer/_api.py

```diff
@@ -335,26 +335,35 @@ def _cohort_list_details(cohort_dict: dict[str, tuple[Any]]) -> HTML:
     from .data.filter import filter_rule_from_cohort_dictionary
 
     sg = Seismogram()
+    cfg = sg.config
+    target_cols = [pdh.event_value(x) for x in cfg.targets]
+    intervention_cols = [pdh.event_value(x) for x in cfg.interventions]
+    outcome_cols = [pdh.event_value(x) for x in cfg.outcomes]
+
     rule = filter_rule_from_cohort_dictionary(cohort_dict)
-    data = rule.filter(sg.dataframe)
+    data = rule.filter(sg.dataframe)[
+        cfg.entity_keys + cfg.output_list + intervention_cols + outcome_cols + target_cols
+    ]
     cohort_count = data[sg.entity_keys[0]].nunique()
     if cohort_count < sg.censor_threshold:
         return template.render_censored_plot_message(sg.censor_threshold)
 
-    cfg = sg.config
-    target_cols = [pdh.event_value(x) for x in cfg.targets]
-    intervention_cols = [pdh.event_value(x) for x in cfg.interventions]
-    outcome_cols = [pdh.event_value(x) for x in cfg.outcomes]
-    groups = data[cfg.entity_keys + cfg.output_list + intervention_cols + outcome_cols + target_cols].groupby(
-        target_cols
-    )
-    aggregation = {cfg.entity_id: ["count", "nunique"]}
-    if len(cfg.context_id):
-        aggregation[cfg.context_id] = "nunique"
-    # add in other keys for aggregation
-    aggregation.update({k: "mean" for k in cfg.output_list + intervention_cols + outcome_cols})
+    groups = data.groupby(target_cols)
+    float_cols = list(data[intervention_cols + outcome_cols].select_dtypes(include=float))
+
+    stat_dict = {k: ["mean"] for k in float_cols}
+    stat_dict.update({cfg.entity_id: ["nunique", "count"], cfg.context_id: ["nunique"]})
+
+    groupstats = groups.agg(stat_dict)
+    groupstats.columns = [pdh.event_name(x) for x in float_cols] + [
+        f"Unique {cfg.entity_id}",
+        f"{cfg.entity_id} Count",
+        f"Unique {cfg.context_id}",
+    ]
+    groupstats.index.rename([pdh.event_name(x) for x in target_cols], inplace=True)
```

Comment:
This line is still creating issues.
MultiIndex (multiple defined targets) seems to work, but it looks like it fails if only one target is defined.  I think rename on single index can't handle being passed a list (my suspicion is its due to pd.Index.name vs pd.MultiIndex.names)

Reply:
fixed, when we move this to use `great_tables`, or move this out of `_api` we should make adding unit tests around it a requirement.